### PR TITLE
Add slice ergonomics.

### PR DESCRIPTION
**Pull request by ry**
👋😊

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-core/964)

### CODE REVIEW

File: src/ops/slice_test.ts [hunk 1]
```diff
@@ -123,6 +123,43 @@ describeWithFlags('slice3d', ALL_ENVS, () => {
   });
 });
 
+describeWithFlags('slice ergonomics', ALL_ENVS, () => {
```
Comment (ry) [inline src/ops/slice_test.ts]: `CPU_ENVS`?

**Comment by dsmilkov** [comment]
:lgtm_strong: Thanks Ryan! Couple small comments.

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/ops/slice.ts, line 100 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/964#-LAOulen0pXeOgIRNdK6:-LAOulen0pXeOgIRNdK7:bxfzl8b) ([raw file](https://github.com/tensorflow/tfjs-core/blob/57cb6cbf351e452bd94658de490f8da0ebf38d84/src/ops/slice.ts#L100)):*
> ```XML
>    * @param x The input `Tensor` to slice from.
>    * @param begin The coordinates to start the slice from. The length of this
>    *     array should match the rank of `x`. Can also be a single number, in
> ```

Change the doc to say something like: The length can be less than the rank of `x` - the rest of the axes will have implicit 0 as start. Can also be a single number, in which case it specifies first axis.

Same for the param `size` below.

---

*[src/ops/slice.ts, line 115 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/964#-LAOuNA8-yAQq5U3Br4J:-LAOuNA8-yAQq5U3Br4K:buq5mlt) ([raw file](https://github.com/tensorflow/tfjs-core/blob/57cb6cbf351e452bd94658de490f8da0ebf38d84/src/ops/slice.ts#L115)):*
> ```XML
>     // The following logic allows for more ergonomic calls.
>     let begin_: number[];
>     if (typeof begin === "number") {
> ```

single quotes `'` for strings for consistency, here and below. Not sure why our linter didn't catch that.

---

*[src/ops/slice_test.ts, line 126 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/964#-LALV45y3q9wQQxvhWo8-r1-126:-LAOvECu2YBplsi9HWPA:btvce4x) ([raw file](https://github.com/tensorflow/tfjs-core/blob/57cb6cbf351e452bd94658de490f8da0ebf38d84/src/ops/slice_test.ts#L126)):*
<details><summary><i>Previously, ry (Ryan Dahl) wrote…</i></summary><blockquote>

Maybe use `CPU_ENVS`? Do these need to be run on every platform?
</blockquote></details>

CPU_ENVS makes sense. Thanks!

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/964#-:-LAOwP4P4zLhYXXm7lcT:b-jny0tk)*

**Comment by ry** [comment]
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[src/ops/slice.ts, line 100 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/964#-LAOulen0pXeOgIRNdK6:-LAPK-WcDfPpgSx-4T7O:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-core/blob/57cb6cbf351e452bd94658de490f8da0ebf38d84/src/ops/slice.ts#L100)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

Change the doc to say something like: The length can be less than the rank of `x` - the rest of the axes will have implicit 0 as start. Can also be a single number, in which case it specifies first axis.

Same for the param `size` below.
</blockquote></details>

Done.

---

*[src/ops/slice.ts, line 115 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/964#-LAOuNA8-yAQq5U3Br4J:-LAPKtDgCaz7ati62gjD:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-core/blob/57cb6cbf351e452bd94658de490f8da0ebf38d84/src/ops/slice.ts#L115)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

single quotes `'` for strings for consistency, here and below. Not sure why our linter didn't catch that.
</blockquote></details>

Done.

---

*[src/ops/slice_test.ts, line 126 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/964#-LALV45y3q9wQQxvhWo8-r1-126:-LAPKu3A4TTRnkuY63Ic:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-core/blob/57cb6cbf351e452bd94658de490f8da0ebf38d84/src/ops/slice_test.ts#L126)):*
<details><summary><i>Previously, dsmilkov (Daniel Smilkov) wrote…</i></summary><blockquote>

CPU_ENVS makes sense. Thanks!
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/964)*

**Comment by dsmilkov** [comment]
:lgtm_strong: Thank you!

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-core/964#-:-LAPQgmz4LCPwe2IEdTv:b-vn5y6t)*

**Pull request merged**